### PR TITLE
 Prevent r2 injection when opening a file

### PR DESCRIPTION
**Pull request by kernelm0de**
When opening a file from File->Open menu the file name is passed to cmd() without validation which can lead to execution of various commands. I have added some code to prevent the above behaviour.

### CODE REVIEW

**Comment by Maijin** [comment]
image

**Comment by Maijin** [comment]
Well my point was, that those filenames are valid and cutter should still be able to open those like other tools do.

**Comment by xarkes** [comment]
Hello @kernelm0de, any update on this?

**Comment by kernelm0de** [comment]
Will fix this soon, got caught up in some stuff.

File: src/Cutter.cpp [hunk 1]
```diff
@@ -397,10 +397,14 @@ bool CutterCore::tryFile(QString path, bool rw)
...
-        cmd("o " + path);
+    CORE_LOCK();
+    RVA addr = mapaddr != RVA_INVALID ? mapaddr : 0;
+    ut64 baddr = Core()->getFileInfo().object()["bin"].toObject()["baddr"].toVariant().toULongLong();
+    if(r_core_file_open(core_, path.toUtf8().constData(), R_PERM_RX, addr)){
+        r_core_bin_load(core_, path.toUtf8().constData(), baddr);
+    }
+    else{
+        eprintf("Cannot open file\n");
     }
```
Comment (xarkes) [inline src/Cutter.cpp]: Do not use `eprintf`, this comes from radare2 land and it is not convenient for a GUI :)
Better do it like here: https://github.com/radareorg/cutter/blob/b95620907aeb77ec672ae5cd7d4edd19becb7b27/src/dialogs/SaveProjectDialog.cpp#L66

File: src/Cutter.cpp [hunk 2]
```diff
@@ -397,10 +397,14 @@ bool CutterCore::tryFile(QString path, bool rw)
...
 void CutterCore::openFile(QString path, RVA mapaddr)
 {
-    if (mapaddr != RVA_INVALID) {
-        cmd("o " + path + QString(" %1").arg(mapaddr));
-    } else {
-        cmd("o " + path);
+    CORE_LOCK();
+    RVA addr = mapaddr != RVA_INVALID ? mapaddr : 0;
+    ut64 baddr = Core()->getFileInfo().object()["bin"].toObject()["baddr"].toVariant().toULongLong();
+    if(r_core_file_open(core_, path.toUtf8().constData(), R_PERM_RX, addr)){
```
Comment (xarkes) [inline src/Cutter.cpp]: Please add a space after the `if` and at the end of the line between the `)` and `{`

File: src/Cutter.cpp [hunk 3]
```diff
@@ -395,17 +395,18 @@ bool CutterCore::tryFile(QString path, bool rw)
...
-void CutterCore::openFile(QString path, RVA mapaddr)
+bool CutterCore::openFile(QString path, RVA mapaddr)
 {
     CORE_LOCK();
     RVA addr = mapaddr != RVA_INVALID ? mapaddr : 0;
     ut64 baddr = Core()->getFileInfo().object()["bin"].toObject()["baddr"].toVariant().toULongLong();
-    if(r_core_file_open(core_, path.toUtf8().constData(), R_PERM_RX, addr)){
+    if (r_core_file_open(core_, path.toUtf8().constData(), R_PERM_RX, addr)) {
         r_core_bin_load(core_, path.toUtf8().constData(), baddr);
     }
```
Comment (ITAYC0HEN) [inline src/Cutter.cpp]: Formatting:

```
if (... ) {

} else {

} 
```

File: src/Cutter.h [hunk 4]
```diff
@@ -449,7 +449,7 @@ class CutterCore: public QObject
     bool loadFile(QString path, ut64 baddr = 0LL, ut64 mapaddr = 0LL, int perms = R_PERM_R,
                   int va = 0, bool loadbin = false, const QString &forceBinPlugin = QString());
     bool tryFile(QString path, bool rw);
-    void openFile(QString path, RVA mapaddr);
+    bool openFile(QString path, RVA mapaddr);
```
Comment (ITAYC0HEN) [inline src/Cutter.h]: Please document this function using the QDocs way
With \brief \params and \returns

File: src/dialogs/OpenFileDialog.cpp [hunk 5]
```diff
@@ -29,7 +29,12 @@ void OpenFileDialog::on_buttonBox_accepted()
     if (!mapAddressStr.isEmpty()) {
         mapAddress = Core()->math(mapAddressStr);
     }
-    Core()->openFile(filePath, mapAddress);
+
+    if (!Core()->openFile(filePath, mapAddress)) {
```
Comment (ITAYC0HEN) [inline src/dialogs/OpenFileDialog.cpp]: Loop until success or cancel

**Pull request merged**